widgetastic_manageiq/__init__.py

```diff
@@ -4207,3 +4207,153 @@ def data_for_timestamp(self, timestamp):
+    """Multi Select Box to be used in Infra Planning Wizard for v2v.
+
+    Args:
+        div_id (str): id of div element that contains multiselectbox
+
+    Usage:
+        e.g. target_clusters = MultiSelectList('target_clusters')
+    """
+    ROOT = ParametrizedLocator('.//div[contains(@id,{@div_id|quote})]')
+    ITEMS_LOCATOR = './div/div/div/span[contains(@class,"dual-pane-mapper-item-container")]'
+    SELECTED_ITEMS_LOCATOR = ('./div[contains(@class,"selected")]/div/div'
+    '/span[contains(@class,"dual-pane-mapper-item-container")]')
+
+    def __init__(self, parent, div_id, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.div_id = div_id
+
+    @property
+    def all_items(self):
+        """Returns list of all items."""
+        return [self.browser.text(item) for item in self.all_items_locators]
+
+    @property
+    def item_count(self):
+        """Returns item count."""
+        return len(self.all_items)
+
+    @property
+    def list_selected_items(self):
+        """Returns list of items that are currently selected."""
+        return [self.browser.text(item) for item in self.list_selected_items_locators]
+
+    @property
+    def all_items_locators(self):
+        """Returns list of item locators for each of the items."""
+        return self.browser.elements(self.ITEMS_LOCATOR)
+
+    @property
+    def list_selected_items_locators(self):
+        """Returns list of item locators for currently selected items."""
+        return self.browser.elements(self.SELECTED_ITEMS_LOCATOR)
+
+    def _get_all_options(self):
+        """ Method to return dict of all the item webelement locators and text"""
+        return {self.browser.text(element): element for element in self.all_items_locators}
+
+    def read(self):
+        """Method to read elements that are currently selected."""
+        return self.list_selected_items
+
+    def fill(self, values, partial_match=False):
+        """Click/Select multiple the items in the multiselectbox
+
+        Args:
+            items: 'list' of items of type 'str' to be matched
+            partial_match: if you want to use partial text matching. Default:False
+
+        Returns:
+            True if all values were found and clicked.
+        """
+        # TODO self.browser.plugin.ensure_page_safe() should be here
+        wait_for(lambda: "loading" not in self.browser.classes(self), timeout=5)
+        if values == self.list_selected_items or values == []:
+            return False
+        if values:
+            for item in values:
+                self.select_by_text(item, partial_match=partial_match)
+        return True
+
+    def select_by_text(self, text, partial_match=False):
+        """Select/click option using text
+
+        Args:
+            text(str): text to be matched
+            partial_match: if you want to use partial text matching. Default:False
+        """
+        try:
+            if text and not partial_match:
```

Comment:
**Required**

I think this should integrate with widgetastic.core.utils.partial_match and a contains xpath, like widgetastic.patternfly.BootstrapNav.

This might be too much to scope for including today though :/